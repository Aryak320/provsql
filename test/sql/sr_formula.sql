\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

/* Example of provenance evaluation */
SELECT create_provenance_mapping('personnel_name', 'personnel', 'name');
CREATE TABLE result_formula AS SELECT
  p1.city,
  sr_formula(provenance(), 'personnel_name') AS formula
FROM personnel p1, personnel p2
WHERE p1.city = p2.city AND p1.id < p2.id
GROUP BY p1.city
ORDER BY p1.city;

SELECT remove_provenance('result_formula');
SELECT
  city,
  replace(formula,'(Dave ⊗ Nancy) ⊕ (Dave ⊗ Magdalen)','(Dave ⊗ Magdalen) ⊕ (Dave ⊗ Nancy)') AS formula
FROM result_formula;

DROP TABLE result_formula;
