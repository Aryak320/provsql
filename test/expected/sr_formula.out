\set ECHO none
remove_provenance

(1 row)
city|formula
Berlin|(Ellen ⊗ Susan)
New York|(John ⊗ Paul)
Paris|((Dave ⊗ Magdalen) ⊕ (Dave ⊗ Nancy) ⊕ (Magdalen ⊗ Nancy))
(3 rows)
